### PR TITLE
C365598 Unlink "MARC Bibliographic" field from "MARC Authority" record (Spitfire) (TaaS)

### PR DESCRIPTION
**Pull request by alex-ivashnev**
[C365598](https://foliotest.testrail.io/index.php?/cases/view/365598)
[Allure report - CI](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/4745/allure)

### CODE REVIEW

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    tag010: '010',
+    tag700: '700',
+    authority100FieldValue: 'C365598 Chin, Staceyann, 1972-',
+    authority010FieldValue: 'n2008052404',
+    authority700FieldValue: 'C365598 Woodson, Jacqueline',
+    successMsg:
+      'This record has successfully saved and is in process. Changes may not appear immediately.',
+    accordion: 'Contributor',
+  };
+  let createdInstanceID;
```
Comment (dnovikau) [inline cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js:26]: Please remove unused constants

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      Permissions.uiQuickMarcQuickMarcBibliographicEditorAll.gui,
+      Permissions.uiMarcAuthoritiesAuthorityRecordView.gui,
+      Permissions.uiQuickMarcQuickMarcAuthoritiesEditorAll.gui,
+      Permissions.moduleDataImportEnabled.gui,
+      Permissions.uiCanLinkUnlinkAuthorityRecordsToBibRecords.gui,
+      Permissions.dataExportEnableApp.gui,
+    ]).then((createdUserProperties) => {
+      testData.userProperties = createdUserProperties;
+
+      cy.login(testData.userProperties.username, testData.userProperties.password, {
```
Comment (dnovikau) [inline cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js:85]: There is no need to perform login as temp user and logining as admin right after is

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  };
+  const contributors = {
+    firstName: 'C365598 Chin, Staceyann, 1972-',
+    secondName: 'Woodson, Jacqueline',
+  };
+  const createdAuthorityIDs = [];
+
+  before(() => {
+    cy.createTempUser([
+      Permissions.inventoryAll.gui,
```
Comment (dnovikau) [inline cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js:73]: Permissions set is not correct, should be:
```JS
      Permissions.inventoryAll.gui,
      Permissions.uiMarcAuthoritiesAuthorityRecordView.gui,
      Permissions.uiQuickMarcQuickMarcAuthoritiesEditorAll.gui,
      Permissions.uiQuickMarcQuickMarcBibliographicEditorAll.gui,
      Permissions.uiQuickMarcQuickMarcAuthorityLinkUnlink.gui,
```

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      Permissions.dataExportEnableApp.gui,
+    ]).then((createdUserProperties) => {
+      testData.userProperties = createdUserProperties;
+
+      cy.login(testData.userProperties.username, testData.userProperties.password, {
+        path: TopMenu.dataImportPath,
+        waiter: DataImport.waitLoading,
+      });
+      marcFiles.forEach((marcFile) => {
+        cy.loginAsAdmin({ path: TopMenu.dataImportPath, waiter: DataImport.waitLoading }).then(
```
Comment (dnovikau) [inline cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js:90]: I'd propose not to make multiple logins but perform login one time and navigate to start page in the loop. Please look at the next example: 
```JS
      cy.loginAsAdmin().then(() => {
        marcFiles.forEach((marcFile) => {
          cy.visit(TopMenu.dataImportPath);
          DataImport.waitLoading();

          DataImport.uploadFile(marcFile.marc, marcFile.fileName);
          JobProfiles.waitLoadingList();
          JobProfiles.searchJobProfileForImport(marcFile.jobProfileToRun);
          JobProfiles.runImportFile();
          JobProfiles.waitFileIsImported(marcFile.fileName);
          Logs.checkStatusOfJobProfile('Completed');
          Logs.openFileDetails(marcFile.fileName);
          for (let i = 0; i < marcFile.numOfRecords; i++) {
            Logs.getCreatedItemsID(i).then((link) => {
              createdAuthorityIDs.push(link.split('/')[5]);
            });
          }
        });
      });
```

File: cypress/support/fragments/inventory/inventoryInstance.js [hunk 5]
```diff
@@ -977,4 +989,24 @@ export default {
...
+        .find(MultiColumnListRow({ index: indexRow }))
+        .find(marcAuthorityAppIcon)
+        .exists(),
+    );
+  },
+
+  checkMarcAppIconAbsent: (indexRow) => {
+    cy.expect(
+      contributorAccordion
+        .find(MultiColumnList({ id: 'list-contributors' }))
```
Comment (dnovikau) [inline cypress/support/fragments/inventory/inventoryInstance.js:1006]: Please create a const for 'MultiColumnList({ id: 'list-contributors' })' interactor as it is used multiple times

**Review by dnovikau — CHANGES_REQUESTED** [review]
Please resolve merge conflicts.

Please find out root cause of the failure (The test is failing both on CI and locally)

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js (hunk 1 above)
Comment (alex-ivashnev) [reply cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js]: resolved

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js (hunk 3 above)
Comment (alex-ivashnev) [reply cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js:72]: resolved

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js (hunk 4 above)
Comment (alex-ivashnev) [reply cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js]: resolved

File: cypress/support/fragments/inventory/inventoryInstance.js (hunk 5 above)
Comment (alex-ivashnev) [reply cypress/support/fragments/inventory/inventoryInstance.js]: resolved

**Review by dnovikau — APPROVED** [review]

File: cypress/support/fragments/marcAuthority/marcAuthority.js [hunk 6]
```diff
@@ -113,6 +113,22 @@ export default {
         .fillIn(content),
     ]);
   },
+  checkFieldByRow: (rowIndex, field) => {
```
Comment (Sherzod-Kenjaev) [inline cypress/support/fragments/marcAuthority/marcAuthority.js:116]: Can you use QuickMarcEditor.verifyTagFieldAfterUnlinking() except creating new one.

File: cypress/support/fragments/marcAuthority/marcAuthority.js [hunk 7]
```diff
@@ -122,6 +138,10 @@ export default {
     cy.expect(buttonLink.exists());
     cy.expect(Callout('Field 700 has been linked to a MARC authority record.').exists());
   },
+  checkLinkingAuthorityByTag: (tag) => {
```
Comment (Sherzod-Kenjaev) [inline cypress/support/fragments/marcAuthority/marcAuthority.js:141]: Create quickmarc editing methods in quickMarcEditor.js file, but before check there please there are methods already created for this type of actions.

File: cypress/support/fragments/marcAuthority/marcAuthority.js [hunk 8]
```diff
@@ -138,6 +158,9 @@ export default {
   clicksaveAndCloseButton: () => {
     cy.do(saveAndCloseButton.click());
   },
+  clickUnlinkButton: () => {
```
Comment (Sherzod-Kenjaev) [inline cypress/support/fragments/marcAuthority/marcAuthority.js:161]: Create quickmarc editing methods in quickMarcEditor.js file, but before check there please there are methods already created for this type of actions.

File: cypress/support/fragments/quickMarcEditor.js [hunk 9]
```diff
@@ -453,6 +456,15 @@ export default {
     cy.do(unlinkModal.find(unlinkButtonInsideModal).click());
   },
 
+  checkUnlinkModal(rowIndex, text) {
+    cy.do(QuickMarcEditorRow({ index: rowIndex }).find(unlinkIconButton).click());
+    cy.expect(unlinkModal.exists());
+    cy.expect(unlinkButtonInsideModal.exists());
+    cy.expect(cancelUnlinkButtonInsideModal.exists());
+    cy.expect(unlinkModal.has({ content: including(text) }));
```
Comment (Sherzod-Kenjaev) [inline cypress/support/fragments/quickMarcEditor.js:464]: Move all above cy.expects checks to array -> cy.expect([])

**Review by Sherzod-Kenjaev — CHANGES_REQUESTED** [review]

File: cypress/support/fragments/marcAuthority/marcAuthority.js (hunk 7 above)
Comment (alex-ivashnev) [reply cypress/support/fragments/marcAuthority/marcAuthority.js:141]: All same methods in file have hardcoded row index in message.

File: cypress/support/fragments/quickMarcEditor.js (hunk 9 above)
Comment (alex-ivashnev) [reply cypress/support/fragments/quickMarcEditor.js:464]: fixed

File: cypress/support/fragments/marcAuthority/marcAuthority.js (hunk 8 above)
Comment (alex-ivashnev) [reply cypress/support/fragments/marcAuthority/marcAuthority.js:161]: fixed

File: cypress/support/fragments/marcAuthority/marcAuthority.js (hunk 7 above)
Comment (alex-ivashnev) [reply cypress/support/fragments/marcAuthority/marcAuthority.js:141]: fixed

File: cypress/support/fragments/marcAuthority/marcAuthority.js (hunk 6 above)
Comment (alex-ivashnev) [reply cypress/support/fragments/marcAuthority/marcAuthority.js:116]: fixed

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+    });
+  });
+
+  it(
+    'C365598 Unlink "MARC Bibliographic" field from "MARC Authority" record and use the "Save & close" button in editing window. (spitfire) (TaaS)',
+    { tags: [TestTypes.criticalPath, DevTeams.spitfire] },
+    () => {
+      InventoryInstance.searchByTitle(createdAuthorityIDs[2]);
+      InventoryInstances.selectInstance();
+      cy.wait(1000);
```
[the remark this answers is not in the document]
Comment (alex-ivashnev) [reply cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js:123]: Done. 2 out of 5 tests failed without this waiter

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+  });
+
+  beforeEach('Login to the application', () => {
+    cy.login(testData.userProperties.username, testData.userProperties.password, {
+      path: TopMenu.inventoryPath,
+      waiter: InventoryInstances.waitContentLoading,
+    });
+  });
+
+  afterEach(() => {
```
[the remark this answers is not in the document]
Comment (alex-ivashnev) [reply cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js:109]: done

File: cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+            Logs.getCreatedItemsID(i).then((link) => {
+              createdAuthorityIDs.push(link.split('/')[5]);
+            });
+          }
+        });
+      });
+    });
+  });
+
+  beforeEach('Login to the application', () => {
```
[the remark this answers is not in the document]
Comment (alex-ivashnev) [reply cypress/e2e/marc-authority/marc-bibliographic-unlink-fields.cy.js:102]: done

**Review by Sherzod-Kenjaev — APPROVED** [review]

**Pull request merged**